### ISSUE TITLE
extract_centerline() should output the real centerline coordinate, not the discrete version

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1938]
### Description

Currently, the text output of extract_centerline() is simply the voxel coordinates of the output image centerline. There is a lack of precision when doing that. We should instead output the floating point coordinates.

**Comment by zougloub** [comment]
TL;DR: I'm wondering what is the convention for real vs. integer coordinates in these images (**pixel-center convention**) and if it's worthwhile to document it.

I'm assuming a coordinate index `idx` corresponds to a real coordinate span of `[idx,idx+1)`, ie. "(z.0,y.0,x.0)" coordinate indicates the corner of a voxel, and a ("z.5", "y.5", "x.5) one indicates the center (**half-integer pixel-center convention**), so that (0,0,0) and (NZ, NY, NX) are corners of a volume (so `int_to_float(x) = x+0.5` and `float_to_int(x) = int(round(x-.5)) = int(x)`).

The alternative I would be less comfortable with would be to have voxel centers on integer coordinates, where the volume boundaries are not as "meaningful".

And I was wondering if this is obvious / common knowledge / or documented somewhere (eg. the convention is documented [in opengl](https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_fragment_coord_conventions.txt)).

**Comment by zougloub** [comment]
Note: [looks like](http://nipy.org/nibabel/coordinate_systems.html#voxel-coordinates-are-in-voxel-space) nibabel's physical <-> pixel position transform is using the integer pixel center convention.

**Comment by zougloub** [comment]
The used convention (integer pixel center) is documented, so closing this parenthesis.

**Comment by zougloub** [comment]
Yep.
